src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -538,6 +544,60 @@ func (c controller) addRouterCAToClusterCA() bool {
 
 }
 
+// TODO: remove
+type OLMOperator struct {
+	name      string
+	namespace string
+	enabled   bool
+	status    string
+}
+
+// TODO: remove
+var olms = map[string]OLMOperator{
+	"lso": {
+		"local-storage-operator.4.6.0-202101300140.p0", "openshift-local-storage", true, "",
+	},
+}
+
+func (c controller) getOLMOperators() ([]OLMOperator, error) {
+	ret := []OLMOperator{}
+	operators, err := c.ic.GetClusterOLM(context.TODO())
+	if err != nil {
+		c.log.WithError(err).Warningf("Failed to connect to assisted service")
+		return ret, err
+	}
+	for _, operator := range *operators {
+		if *operator.Enabled && operator.Status != "Succeeded" {
+			ret = append(ret, olms[string(operator.OperatorType)])
+		}
+	}
+	return ret, nil
+}
+
+// TODO: remove
+func updateStatus(olm OLMOperator, phase string) {
+	olm.status = phase
+	olms[olm.name] = olm
+}
+
+func (c controller) waitForOLMOperators() bool {
+	c.log.Infof("Checking OLM operators")
+	opeartors, _ := c.getOLMOperators()
+	if len(opeartors) == 0 {
+		return true
+	}
+	for _, olm := range opeartors {
+		csv, err := c.kc.GetCSV(olm.namespace, olm.name)
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to get %s", olm.name)
+		}
+		// TODO: c.ic.UpdateClusterOLM(context.TODO(), csv.name)
+		updateStatus(olm, string(csv.Status.Phase)) // TODO: remove
+		c.log.Infof("CSV %s is in status %s.", olm.name, csv.Status.Phase)
+	}
+	return false
```

Comment:
I'm confused why are you returning `false` here. I'd expect the return value to depend on the status of the CSV of all operators: if all CSVs have a succeeded state, then the return should be true so that the call in line 293 will continue successfully, else false.
If I'm reading this correctly, this will always return `false` regardless of the status of the CSVs, and only `true` when there are no OLM operators to install.

Reply:
Actually this condition is handled on line 587. `getOLMOperators` should return operators which are still installing, and if there are all in succeded state, we return `true`. If there are some operators not yet in `Succeeded` state we return false and timeout eventually if needed.

Reply:
Maybe rename `getOLMOperators` to `getPendingOLMOperators` or something more meaningful? If `getOLMOperators` return values depend on the status of the CSV, the current name might lead to confusion to think that the output of the command will always return the list of selected operators (as I wrongly expected), regardless of their deployment status. WDYT?

Reply:
Makes sense, thanks for review.